components/layout_2020/query.rs

```diff
@@ -437,9 +443,38 @@ fn process_offset_parent_query_inner(
     // zero and terminate this algorithm." (others)
     let node_offset_box = node_offset_box?;
 
-    let offset_parent_padding_box_corner = node_offset_box
-        .offset_parent_node_address
-        .map(|offset_parent_node_address| {
+    let offset_parent_padding_box_corner = if let Some(offset_parent_node_address) =
+        node_offset_box.offset_parent_node_address
+    {
+        // Although the spec (https://www.w3.org/TR/cssom-view-1/#extensions-to-the-htmlelement-interface)
+        // says that offsetTop/offsetLeft are always relative to the padding box of the offsetParent.
+        // In practice this is not true in major browsers in the case that the offsetParent is the body
+        // element and the body element is position:static. In that case offsetLeft/offsetTop are computed
+        // relative to the root nodes border box.
+        if node_offset_box.is_static_body_element {
+            fn extract_box_fragment(
+                fragment: &Fragment,
+                containing_block: &euclid::Rect<Au, CSSPixel>,
+            ) -> Vector2D<Au, UnknownUnit> {
+                let (Fragment::Box(fragment) | Fragment::Float(fragment)) = fragment else {
+                    unreachable!();
+                };
+                // Again, take the *first* associated CSS layout box.
+                let border_box_corner =
+                    fragment.border_rect().origin.to_vector() + containing_block.origin.to_vector();
+                border_box_corner.to_untyped()
+            }
```

Comment:
Nit: no need to import `UnknownUnit` if using a lambda. And `euclid::Rect<Au, CSSPixel>` can be `PhysicalRect<Au>` .

```suggestion
            let extract_box_fragment =
                |fragment: &Fragment, containing_block: &PhysicalRect<Au>| {
                    let (Fragment::Box(fragment) | Fragment::Float(fragment)) = fragment else {
                        unreachable!();
                    };
                    // Again, take the *first* associated CSS layout box.
                    let border_box_corner = fragment.border_rect().origin.to_vector() +
                        containing_block.origin.to_vector();
                    border_box_corner.to_untyped()
                };
```